raiden-ts/src/transfers/epics/utils.ts

```diff
@@ -108,3 +111,48 @@ export function matchWithdraw<
           'totalWithdraw' in data ? data.totalWithdraw : data.total_withdraw,
         )));
 }
+
+/**
+ * @param channel - Channel state
+ * @param config - Raiden config
+ * @returns FeeSchedule for channel
+ */
+export function getFeeScheduleForChannel(
+  channel: Pick<Channel, 'token'>,
+  config: Pick<RaidenConfig, 'tokenFees'>,
+): FeeSchedule {
+  const tokenSchedule = {
+    flat: Zero as Int<32>,
+    ...config.tokenFees[AddressZero],
+    ...config.tokenFees[channel.token],
+  };
+  // split per-token feeSchedule in half, one for each channel (in & out)
+  const channelSchedule = {
+    ...tokenSchedule,
+    flat: tokenSchedule.flat.div(2) as Int<32>,
+  };
+  return channelSchedule;
+}
+
+/**
+ * Calculates the mediation fees for given input amount and channel pairs
+ *
+ * @param _amountIn - Incoming transfer amount
+ * @param channelIn - Incoming channel state
+ * @param channelOut - Outgoing channel state
+ * @param opts - Options object
+ * @param opts.config - Config object
+ * @returns Fee to be taken from input: output = input - fee
+ */
+export function calculateMediationFee(
+  _amountIn: UInt<32>,
+  channelIn: Parameters<typeof getFeeScheduleForChannel>[0],
+  channelOut: Parameters<typeof getFeeScheduleForChannel>[0],
+  { config }: { config: Parameters<typeof getFeeScheduleForChannel>[1] },
+): Int<32> {
+  const feeScheduleIn = getFeeScheduleForChannel(channelIn, config);
+  const feeScheduleOut = getFeeScheduleForChannel(channelOut, config);
+
+  const flatFee = feeScheduleIn.flat.add(feeScheduleOut.flat);
```

Comment:
> This code must not know which models actually exist.

You're again coming from the perspective that these fee models should/could be a plugin, which is not present on any place of the requirements nor the reference implementation.
From my perspective, this code is quite extensible already: a developer in need to change/add fee models have just two places to change: the config options they need/want, and this function which calculates the actual fee.
Unfortunately, per protocol, I think what you propose is not even possible, since the `imbalance` fees actually depend on the other values to be calculated, so the best that I could do is what I did: isolate it in this single, pure function, which returns an obvious result: the actual, final fee for that transfer.